tests/foreman/api/test_repositories.py

```diff
@@ -199,10 +198,9 @@ def test_negative_upload_expired_manifest(module_org, target_sat):
 
     :expectedresults: Manifest refresh should fail and return error message
     """
-    manifester = Manifester(manifest_category=settings.manifest.entitlement)
-    manifest = manifester.get_manifest()
+    manifest = function_entitlement_manifest.get_manifest()
```

Comment:
@dosas As you've mentioned `it does not return an instance of manifester but the result of get_manifest()`, so what if we can just use function_entitlement_manifest for manifest? 
CC @synkd @jyejare wdyt?
```suggestion
    target_sat.upload_manifest(module_org.id, function_entitlement_manifest.content)
    request.addfinalizer(function_entitlement_manifest.delete_subscription_allocation)
```

Reply:
@Gauravtalreja1 This would be technically possible, but currently, it's a bit simpler to generate the manifest from a local Manifester instance. Manifester's `delete_subscription_allocation` method does allow passing in an arbitrary UUID argument for deletion [1], but the allocation UUID is not immediately accessible from the `function_entitlement_manifest` object. It's embedded in `function_entitlement_manifest.url` but would need to be extracted from a URL path like this one [2]. 

With that said, it should be a fairly simple change to add a line like the following to the end of Manifester's `trigger_manifest_export` method [3]:
```
manifest.uuid = self.allocation_uuid
``` 
The UUID would then be accessible as `function_entitlement_manifest.uuid`, which could be passed to `Manifester.delete_subscription_allocation()`. From the looks of it, this same change would also enable `tests/upgrades/test_subscription.py::test_post_subscription_auto_attach` to use a Manifester fixture.

If this is the direction that people want to go, then I'll also note that I would welcome a community contribution to Manifester making the necessary change.

[1] https://github.com/SatelliteQE/manifester/blob/8ff1e296d5dff89ccd9d76deffca8592807064ff/manifester/manifester.py#L184

[2] 'https://api.access.redhat.com/management/v1/allocations/0df90e19-9fda-49d8-8a3f-c7bc29ef1b23/export/2c94e7b58e14afed018e43cacf5b4db9'

[3] https://github.com/SatelliteQE/manifester/blob/8ff1e296d5dff89ccd9d76deffca8592807064ff/manifester/manifester.py#L400

Reply:
@synkd I can continue here once a new release of manifester is out

Reply:
@dosas I just cut a new Manifester release that includes this change. However, the build/publish to PyPi job failed, and I will be OOO the rest of this week and next week. We'll need to hold off on merging this change until the new Manifester release is published, which may need to wait until I return.

Reply:
I doubt that the uuid change will enable the usage of the fixture since it still does not provide an instance of manifester

Reply:
@synkd hello, mind revisiting this? it's been sitting here for a while. Thanks!